### ISSUE TITLE
Continuing development for this extention?

### ISSUE THREAD

**Issue by raymondclowe** [bakaburg1/AiQuickFill#1]
Hi, this extension idea looks great and is something I've been looking for for a while now. None of the other "AI writing" tools seem to focus on the idea of suggesting more words as you type, which is what I'm looking for.

Are you continuing the development of the extension? I do have some ideas for additions. Although my JS is pretty basic I can probably fork and add some of them myself, and PR if that interests you.

Alternatively, if you have a reason for not developing this - perhaps you discovered some flaw in the idea, or perhaps found a different way to accomplish a goal, I'd be interested to hear that as well.

:-)

**Comment by bakaburg1** [comment]
Hi! No the idea is still valid! I just simply do not have the time to keep developing it! I made some local changes that I need to push, but it's not on the top of my list, unfortunately.

One very good improvement would be allowing connection to offline LLMa, since I realized that as it is now it bites in my API cost severely. Now-a-days modern LLM would be good enough for this use case.

I'd be very happy if you want to PR any improvement in!

**Comment by raymondclowe** [comment]
Yes, variable back-ends is something I see the need for. I'm currently looking at Mixtral on replicate.com as a good option. And some other things to control budget such as restricting calls a bit - not on every space, but space+timeout for example. Then cancelling in-progress predictions that have become outdated (as the user has resumed typing). All that should save costs.

Then I need it to work with Wordpress editor, and that isn't a "textarea" but some funky change events on a p tag. :-/ 

I'll have a think about it. Unfortunately my JS skills are really quite primitive, but with the help of AI I might be able to do it! :-)

Maybe you could push any additions you've made, even if just to a dev branch if they don't fully work yet. Would avoid some conflicts if I start to hack around with it then want to merge.

**Comment by bakaburg1** [comment]
Sure, I'll try to do it!
If it's possible to make it communicate with an API on a localhost, then one just needs to run a model via Ollama or LMStudio, and the cost is practically zero (conditional on electricity and a powerful enough computer).

**Comment by raymondclowe** [comment]
It seems this is much more complex than I thought. Particularly because my main use case is Wordpress, and that isn't using text areas or any normal input form.

I've started to develop a Wordpress plugin that serves a JS tool which uses Mistral platform as the back end. I don't have access to a fast enough machine to make localhost usable, though I can see the benefit.

I don't know if I can combine the wordpress functionality into the Chrome extention instead of as a plugin. 

Do other web based interface like Gmail or Google docs also have this issue? They are not textareas either I think.

Does that mean the extention need to simulate keypresses if it is to be generic for any website?
